src/transformers/trainer.py

```diff
@@ -1076,6 +1076,9 @@ def train(
                 # If the model is on the GPU, it still works!
                 self._load_state_dict_in_model(state_dict)
 
+            # release memory
+            del state_dict
```

Comment:
```suggestion
                # release memory
                del state_dict
```
It needs to be indented one more level or `state_dict` does not exist. :-)

Reply:
oops, thank you for noticing!

Reply:
why didn't the tests catch that?

Reply:
do we not have a resume from checkpoint test, or was the logic missed with the recent CI logic?

Reply:
ah, no, this branch always get executed so `state_dict` is always set unless deepspeed. so all is good test wise.